### ISSUE TITLE
A Runnable Dependency and Package Setting

### ISSUE THREAD

**Issue by iamtonymwt** [yang-song/score_sde_pytorch#70]
Hey guys, I found a stable environment that could run this code base. Here are my settings.

System: WSL2 Ubuntu 22.04
GPU: NVIDIA A6000;  NVIDIA-SMI 550.54.10; Driver Version: 551.61; CUDA Version: 12.4
Local: Cuda toolkit: release 12.1, V12.1.66
Python: 3.7.12

PIP List:
absl-py==0.10.0
astunparse==1.6.3
attrs==24.2.0
backcall==0.2.0
beautifulsoup4==4.12.3
cachetools==4.2.4
certifi==2024.8.30
charset-normalizer==3.4.0
cloudpickle==2.2.1
contextlib2==21.6.0
cycler==0.11.0
debugpy==1.6.3
decorator==5.1.1
dill==0.3.7
dm-tree==0.1.8
entrypoints==0.4
et-xmlfile==1.1.0
etils==0.9.0
filelock==3.12.2
flatbuffers==1.12
fonttools==4.38.0
future==1.0.0
gast==0.3.3
gdown==4.7.3
google-auth==1.35.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
googleapis-common-protos==1.65.0
grpcio==1.32.0
h5py==2.10.0
idna==3.10
importlib-metadata==6.7.0
importlib-resources==5.12.0
ipykernel==6.16.2
ipython==7.33.0
jax==0.3.25
jedi==0.19.1
joblib==1.3.2
jupyter_client==7.4.9
jupyter_core==4.11.1
Keras-Preprocessing==1.1.2
kiwisolver==1.4.5
Markdown==3.4.4
MarkupSafe==2.1.5
matplotlib==3.5.3
matplotlip==0.2
ml-collections==0.1.0
nest_asyncio==1.6.0
ninja==1.11.1.1
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
oauthlib==3.2.2
openpyxl==3.1.3
opt-einsum==3.3.0
packaging==23.2
pandas==1.3.5
parso==0.8.4
pexpect==4.9.0
pickleshare==0.7.5
Pillow==9.5.0
pip==24.0
promise==2.3
prompt_toolkit==3.0.48
protobuf==3.20.3
psutil==5.9.3
ptyprocess==0.7.0
pyasn1==0.5.1
pyasn1-modules==0.3.0
Pygments==2.17.2
pyparsing==3.1.4
PySocks==1.7.1
python-dateutil==2.9.0
pytz==2024.2
PyYAML==6.0.1
pyzmq==23.2.0
requests==2.31.0
requests-oauthlib==2.0.0
rsa==4.9
scikit-learn==1.0.2
scipy==1.7.3
seaborn==0.12.2
setuptools==69.0.3
six==1.15.0
soupsieve==2.4.1
tensorboard==2.4.1
tensorboard-plugin-wit==1.8.1
tensorflow==2.4.0
tensorflow-addons==0.12.0
tensorflow-datasets==3.1.0
tensorflow-estimator==2.4.0
tensorflow-gan==2.0.0
tensorflow-hub==0.16.0
tensorflow-io==0.34.0
tensorflow-io-gcs-filesystem==0.34.0
tensorflow-metadata==1.12.0
tensorflow-probability==0.12.2
termcolor==1.1.0
threadpoolctl==3.1.0
torch==1.13.1
torch-tb-profiler==0.4.3
torchvision==0.14.1
tornado==6.2
tqdm==4.67.0
traitlets==5.9.0
typeguard==2.13.3
typing-extensions==3.7.4.3
urllib3==2.0.7
wcwidth==0.2.10
Werkzeug==2.2.3
wheel==0.42.0
wrapt==1.12.1
zipp==3.15.0

After that. run:
pip install jaxlib==0.3.25 -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.htm

Although there are some conflicts, but could run the code base successfully.

**Comment by AlexSm1248** [comment]
Hi iamtonymwt thank you so much for the suggestion, by any chance do you happen to have the numpy version that works for you? I am asking because that seems to be one of the issues for me. 

Also "nvidia-cudnn-cu11==8.5.0.96" gives an error, "Could not find a version that satisfies the requirement nvidia-cudnn-cu11==8.5.0.96"

Thank you,
Alex

**Comment by bhupender-kaushal** [comment]
Hey, but how did you install cuda toolkit 12.1 with torch 1.13 is that supported ?

**Comment by rzjiang0603** [comment]
Thanks for your sharing. Also I can give some additional note here. Seems that jax & jaxlib are the mean reason causing some conflicts (due to the different version requirements of numpy). For those people who are setting the environment, better install other requirements first and then install jax separately.

**Comment by dexin-peng** [comment]
If you encounter some dependency issues, maybe you'll be interested in [my rewrite for Score-based SDE](https://github.com/dexin-peng/score_sde_pytorch), which removed requirements for tensorflow, jax, and is more compatible with torch. A dockerfile is provided.

**Comment by dexin-peng** [comment]
There is plenty of dependency issues. Someone maintaining this repo should consider closing these duplicates.

https://github.com/yang-song/score_sde_pytorch/issues/6 https://github.com/yang-song/score_sde_pytorch/issues/16 https://github.com/yang-song/score_sde_pytorch/issues/23 https://github.com/yang-song/score_sde_pytorch/issues/36 https://github.com/yang-song/score_sde_pytorch/issues/42 https://github.com/yang-song/score_sde_pytorch/issues/61 https://github.com/yang-song/score_sde_pytorch/issues/68 https://github.com/yang-song/score_sde_pytorch/issues/69 https://github.com/yang-song/score_sde_pytorch/issues/70 https://github.com/yang-song/score_sde_pytorch/issues/72

**Comment by Leeyed** [comment]
> Hey guys, I found a stable environment that could run this code base. Here are my settings.
> 
> System: WSL2 Ubuntu 22.04 GPU: NVIDIA A6000; NVIDIA-SMI 550.54.10; Driver Version: 551.61; CUDA Version: 12.4 Local: Cuda toolkit: release 12.1, V12.1.66 Python: 3.7.12
> 
> PIP List: absl-py==0.10.0 astunparse==1.6.3 attrs==24.2.0 backcall==0.2.0 beautifulsoup4==4.12.3 cachetools==4.2.4 certifi==2024.8.30 charset-normalizer==3.4.0 cloudpickle==2.2.1 contextlib2==21.6.0 cycler==0.11.0 debugpy==1.6.3 decorator==5.1.1 dill==0.3.7 dm-tree==0.1.8 entrypoints==0.4 et-xmlfile==1.1.0 etils==0.9.0 filelock==3.12.2 flatbuffers==1.12 fonttools==4.38.0 future==1.0.0 gast==0.3.3 gdown==4.7.3 google-auth==1.35.0 google-auth-oauthlib==0.4.6 google-pasta==0.2.0 googleapis-common-protos==1.65.0 grpcio==1.32.0 h5py==2.10.0 idna==3.10 importlib-metadata==6.7.0 importlib-resources==5.12.0 ipykernel==6.16.2 ipython==7.33.0 jax==0.3.25 jedi==0.19.1 joblib==1.3.2 jupyter_client==7.4.9 jupyter_core==4.11.1 Keras-Preprocessing==1.1.2 kiwisolver==1.4.5 Markdown==3.4.4 MarkupSafe==2.1.5 matplotlib==3.5.3 matplotlip==0.2 ml-collections==0.1.0 nest_asyncio==1.6.0 ninja==1.11.1.1 nvidia-cublas-cu11==11.10.3.66 nvidia-cuda-nvrtc-cu11==11.7.99 nvidia-cuda-runtime-cu11==11.7.99 nvidia-cudnn-cu11==8.5.0.96 oauthlib==3.2.2 openpyxl==3.1.3 opt-einsum==3.3.0 packaging==23.2 pandas==1.3.5 parso==0.8.4 pexpect==4.9.0 pickleshare==0.7.5 Pillow==9.5.0 pip==24.0 promise==2.3 prompt_toolkit==3.0.48 protobuf==3.20.3 psutil==5.9.3 ptyprocess==0.7.0 pyasn1==0.5.1 pyasn1-modules==0.3.0 Pygments==2.17.2 pyparsing==3.1.4 PySocks==1.7.1 python-dateutil==2.9.0 pytz==2024.2 PyYAML==6.0.1 pyzmq==23.2.0 requests==2.31.0 requests-oauthlib==2.0.0 rsa==4.9 scikit-learn==1.0.2 scipy==1.7.3 seaborn==0.12.2 setuptools==69.0.3 six==1.15.0 soupsieve==2.4.1 tensorboard==2.4.1 tensorboard-plugin-wit==1.8.1 tensorflow==2.4.0 tensorflow-addons==0.12.0 tensorflow-datasets==3.1.0 tensorflow-estimator==2.4.0 tensorflow-gan==2.0.0 tensorflow-hub==0.16.0 tensorflow-io==0.34.0 tensorflow-io-gcs-filesystem==0.34.0 tensorflow-metadata==1.12.0 tensorflow-probability==0.12.2 termcolor==1.1.0 threadpoolctl==3.1.0 torch==1.13.1 torch-tb-profiler==0.4.3 torchvision==0.14.1 tornado==6.2 tqdm==4.67.0 traitlets==5.9.0 typeguard==2.13.3 typing-extensions==3.7.4.3 urllib3==2.0.7 wcwidth==0.2.10 Werkzeug==2.2.3 wheel==0.42.0 wrapt==1.12.1 zipp==3.15.0
> 
> After that. run: pip install jaxlib==0.3.25 -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.htm
> 
> Although there are some conflicts, but could run the code base successfully.

Hi, 
it seems a wrong spelling or other mistakes?

ERROR: Could not find a version that satisfies the requirement matplotlip==0.2 (from versions: none)
ERROR: No matching distribution found for matplotlip==0.2
